eclair-core/src/main/scala/fr/acinq/eclair/db/FileBackupHandler.scala

```diff
@@ -16,80 +16,100 @@
-        Metrics.FileBackupCompleted.withoutTags().increment()
-      }
+  sealed trait BackupEvent
+  // this notification is sent when we have completed our backup process (our backup file is ready to be used)
+  case object BackupCompleted extends BackupEvent
+  // @formatter:on
 
-      backupScript_opt.foreach(backupScript => {
-        Try {
-          // run the script in the current thread and wait until it terminates
-          Process(backupScript).!
-        } match {
-          case Success(exitCode) => log.debug(s"backup notify script $backupScript returned $exitCode")
-          case Failure(cause) => log.warning(s"cannot start backup notify script $backupScript:  $cause")
-        }
-      })
-  }
+  def apply(databases: FileBackup, backupParams: FileBackupParams): Behavior[Command] =
+    Behaviors.setup { context =>
+      // we listen to ChannelPersisted events, which will trigger a backup
+      context.system.eventStream ! EventStream.Subscribe(context.messageAdapter[ChannelPersisted](WrappedChannelPersisted))
+      Behaviors.withTimers { timers =>
+        timers.startTimerAtFixedRate(TickBackup, backupParams.interval)
+        new FileBackupHandler(databases, backupParams, context).normal(false)
+      }
+    }
 }
 
-sealed trait BackupEvent
+class FileBackupHandler private(databases: FileBackup,
+                                backupParams: FileBackupParams,
+                                context: ActorContext[Command]) {
 
-// this notification is sent when we have completed our backup process (our backup file is ready to be used)
-case object BackupCompleted extends BackupEvent
+  def normal(willBackup: Boolean): Behavior[Command] =
+    Behaviors.receiveMessagePartial {
+      case _: WrappedChannelPersisted =>
+        context.log.debug("will perform backup")
+        normal(willBackup = true)
+      case TickBackup => if (willBackup) {
+        context.log.debug("performing backup")
+        doBackup()
+        normal(willBackup = false)
+      } else {
+        Behaviors.same
+      }
+    }
 
-object FileBackupHandler {
-  // using this method is the only way to create a BackupHandler actor
-  // we make sure that it uses a custom bounded mailbox, and a custom pinned dispatcher (i.e our actor will have its own thread pool with 1 single thread)
-  def props(databases: FileBackup, backupFile: File, backupScript_opt: Option[String]) =
-    Props(new FileBackupHandler(databases, backupFile, backupScript_opt))
-      .withMailbox("eclair.backup-mailbox")
-      .withDispatcher("eclair.backup-dispatcher")
-}
+  private def doBackup(): Unit = {
+    KamonExt.time(Metrics.FileBackupDuration.withoutTags()) {
+      val tmpFile = new File(backupParams.targetFile.getAbsolutePath.concat(".tmp"))
+      databases.backup(tmpFile)
+
+      // this will throw an exception if it fails, which is possible if the backup file is not on the same filesystem
+      // as the temporary file
```

Comment:
Do we want to catch that and log an error instead of letting the actor crash?
Now that we moved to akka typed, the actor will not be restarted if it crashes since you didn't provide a `SupervisorStrategy` in `Setup`.

Reply:
Fixed with d2a3998fc.